spec/models/moment_spec.rb

```diff
@@ -18,24 +18,44 @@
 #
 
 describe Moment do
-	describe "POST create" do
-		it "create private moment" do
-			new_user = create(:user1)
-			new_category = create(:category, userid: new_user.id)
-			new_mood = create(:mood, userid: new_user.id)
-			new_moment = create(:moment, userid: new_user.id, category: Array.new(1, new_category.id), mood: Array.new(1, new_mood.id))
-			expect(new_moment.viewers).to be_empty
-		  end
+
+	describe '.find_secret_share!' do
+
+		context 'when a valid secret share moment exists' do
+			let(:moment) { create(:moment, :with_user, :with_secret_share) }
+			subject { Moment.find_secret_share!(moment.secret_share_identifier) }
+		  it { is_expected.to eq(moment) }
+		end
+
+		context 'when a secret share moment has expired' do
+			let(:moment) do
+				m = build(:moment, :with_user, :with_secret_share)
+				m.secret_share_expires_at = 1.day.ago
+				m.save!
+				m
+			end
+			
+			subject { Moment.find_secret_share!(moment.secret_share_identifier) }
+			specify { expect { subject }.to raise_error(ActiveRecord::RecordNotFound) }
+		end
+
+		context 'when there is no secret share moment' do
+			subject { Moment.find_secret_share!('foobar') }
+			specify { expect { subject }.to raise_error(ActiveRecord::RecordNotFound) }
+		end
 	end
-	describe "POST create" do
-		it "create public moment" do
-			new_user = create(:user1)
-			new_user2 = create(:user2)
-			new_allies = create(:allyships_accepted, user_id: new_user.id, ally_id: new_user2.id)
-				new_category = create(:category, userid: new_user.id)
-			new_mood = create(:mood, userid: new_user.id)
-			new_moment = create(:moment, userid: new_user.id, category: Array.new(1, new_category.id), mood: Array.new(1, new_mood.id), viewers: [new_user2.id])
-			expect(new_moment.viewers.count).to eq(1)
+
+
```

Comment:
No need to be greedy, one line break is enough.

Reply:
Omg good catch. If you're using Atom or any other IDE, you can remove trailing whitespace through your settings. It does it automagically for you! 🌠